src/osd/ReplicatedPG.cc

```diff
@@ -1651,7 +1651,7 @@ void ReplicatedPG::do_op(OpRequestRef& op)
     // purposes here it doesn't matter which one we get.
     eversion_t replay_version;
     version_t user_version;
-    bool got = pg_log.get_log().get_request(
+    bool got = const_cast<PGLog::IndexedLog&>(pg_log.get_log()).get_request(
```

Comment:
I don't think we want const_cast.

Instead, we could make the indexes in PGLog mutable and leave all those methods const. Or, just make get_log() return a non-const ref...